src/ansys/fluent/core/filereader/casereader.py

```diff
@@ -73,36 +74,53 @@ class CaseReader:
         Get the precision (1 or 2 for 1D of 2D)
     """
 
-    def __init__(self, hdf5_case_filepath: str):
+    def __init__(self, case_filepath: str):
+
         try:
-            file = h5py.File(hdf5_case_filepath)
+            if Path(case_filepath).suffix == ".h5":
+                file = h5py.File(case_filepath)
+                settings = file["settings"]
+                rpvars = settings["Rampant Variables"][0]
+                rp_vars_str = rpvars.decode()
+            elif Path(case_filepath).suffix == ".cas":
+                with open(case_filepath, "rb") as file:
+                    rp_vars_str = file.read()
+                rp_vars_str = get_processed_string(rp_vars_str)
+
+            elif Path(case_filepath).suffix == ".gz":
+                with gzip.open(case_filepath, "rb") as file:
+                    rp_vars_str = file.read()
+                rp_vars_str = get_processed_string(rp_vars_str)
+            else:
+                raise RuntimeError()
+
         except FileNotFoundError:
-            raise RuntimeError(f"The case file {hdf5_case_filepath} cannot be found.")
+            raise RuntimeError(f"The case file {case_filepath} cannot be found.")
         except OSError:
             error_message = (
-                "Could not read case file. " "Only valid HDF5 files can be read. "
+                "Could not read case file. "
+                "Only valid Case files (.h5, .cas, .cas.gz) can be read. "
             )
-            if Path(hdf5_case_filepath).suffix != ".h5":
-                error_message += (
-                    f"The file {hdf5_case_filepath} does not have a .h5 extension."
-                )
             raise RuntimeError(error_message)
+
         except BaseException:
-            raise RuntimeError(f"Could not read case file {hdf5_case_filepath}")
-        settings = file["settings"]
-        rpvars = settings["Rampant Variables"][0]
-        rp_vars_str = rpvars.decode()
+            raise RuntimeError(f"Could not read case file {case_filepath}")
+
         self._rp_vars = lispy.parse(rp_vars_str)[1]
         self._rp_var_cache = {}
```

Comment:
If you want to be more pythonic.  
```suggestion
        self._rp_var_cache = dict()
```